test/pgtest/copy-from.pt

```diff
@@ -0,0 +1,253 @@
+Query {"query": "COPY a FROM STDIN"}
+----
+
+until
+CopyIn
+----
+CopyIn {"format":"text","column_formats":["text","text","text"]}
+
+send
+CopyData "text as int\t2.1\ttext\n"
+CopyDone
+----
+
+until
+ReadyForQuery
+----
+ErrorResponse {"fields":[{"typ":"C","value":"22023"},{"typ":"M","value":"unable to decode column: invalid input syntax for type integer: invalid digit found in string: \"text as int\""}]}
+ReadyForQuery {"status":"I"}
+
+# invalid type
+send
+Query {"query": "COPY a FROM STDIN"}
+----
+
+until
+CopyIn
+----
+CopyIn {"format":"text","column_formats":["text","text","text"]}
+
+send
+CopyData "1\t2.1a\ttext\n"
+CopyDone
+----
+
+until
+ReadyForQuery
+----
+ErrorResponse {"fields":[{"typ":"C","value":"22023"},{"typ":"M","value":"unable to decode column: invalid input syntax for type double precision: \"2.1a\""}]}
+ReadyForQuery {"status":"I"}
+
+#
+# Valid data
+#
+
+# send single valid row
+send
+Query {"query": "COPY a FROM STDIN"}
+----
+
+until
+CopyIn
+----
+CopyIn {"format":"text","column_formats":["text","text","text"]}
+
+send
+CopyData "1\t2.1\ttext\n"
+CopyDone
+----
+
+until
+ReadyForQuery
+----
+CommandComplete {"tag":"COPY 1"}
+ReadyForQuery {"status":"I"}
+
+# send multiple valid rows, multiple messages
+send
+Query {"query": "COPY a FROM STDIN"}
+----
+
+until
+CopyIn
+----
+CopyIn {"format":"text","column_formats":["text","text","text"]}
+
+send
+CopyData "2\t2.1\tsome other text\n"
+CopyData "3\t2.1\tlong text\n"
+CopyDone
+----
+
+until
+ReadyForQuery
+----
+CommandComplete {"tag":"COPY 2"}
+ReadyForQuery {"status":"I"}
+
+# send multiple valid rows, multiple messages, unaligned data
+send
+Query {"query": "COPY a FROM STDIN"}
+----
+
+until
+CopyIn
+----
+CopyIn {"format":"text","column_formats":["text","text","text"]}
+
+send
+CopyData "2\t2.1\tsome other text\n4\t3.1\tlong"
+CopyData " text "
+CopyData "and some more\n"
+CopyDone
+----
+
+until
+ReadyForQuery
+----
+CommandComplete {"tag":"COPY 2"}
+ReadyForQuery {"status":"I"}
+
+# send multiple valid rows, single message
+send
+Query {"query": "COPY a FROM STDIN"}
+----
+
+until
+CopyIn
+----
+CopyIn {"format":"text","column_formats":["text","text","text"]}
+
+send
+CopyData "2\t2.1\tsome other text\n3\t2.1\tlong text\n"
+CopyDone
+----
+
+until
+ReadyForQuery
+----
+CommandComplete {"tag":"COPY 2"}
+ReadyForQuery {"status":"I"}
+
+# valid row with no new line at the end
+send
+Query {"query": "COPY a FROM STDIN"}
+----
+
+until
+CopyIn
+----
+CopyIn {"format":"text","column_formats":["text","text","text"]}
+
+send
+CopyData "5\t5.5\tno new line at the end"
+CopyDone
+----
+
+until
+ReadyForQuery
+----
+CommandComplete {"tag":"COPY 1"}
+ReadyForQuery {"status":"I"}
+
+# dump table
+send
+Query {"query": "COPY (SELECT * FROM a ORDER BY a, b, c) TO STDOUT"}
+----
+
+until
+ReadyForQuery
+----
+CopyOut {"format":"text","column_formats":["text","text","text"]}
+CopyData "1\t2.1\ttext\n"
+CopyData "2\t2.1\tsome other text\n"
+CopyData "2\t2.1\tsome other text\n"
```

Comment:
The handling of ```\.``` is also different when compared to postgres:

In Mz:

```
materialize=> copy t1 from stdin;
Enter data to be copied followed by a newline.
End with a backslash and a period on a line by itself, or an EOF signal.
>> a\.
>> \.
COPY 1
materialize=> select * from t1;
 f1  
-----
 a\.
(1 row)
```
In postgres:

```
pstoev=# select * from t1;
 f1 
----
 a
(1 row)
```